packages/open-payments/src/client/index.ts

```diff
@@ -258,9 +286,23 @@ export interface AuthenticatedClient
   quote: QuoteRoutes
 }
 
-export const createAuthenticatedClient = async (
+/**
+ * @experimental The `requestInterceptor` feature is currently experimental and might be removed in upcoming versions. Use at your own risk! It offers the capability to add a custom method for generating HTTP signatures. It is recommended to create the authenticated client with the `privateKey` and `keyId` arguments. If both `requestInterceptor` and `privateKey`/`keyId` are provided, an error will be thrown.
+ * @throws OpenPaymentsClientError
+ */
+export async function createAuthenticatedClient(
```

Comment:
Even with function overloading, I think we should keep the check. While `createAuthenticatedClientArgs` will support both types without the need for type-narrowing when invoking it, the check will make sure that users fully understand the type of authenticated client they are going to use. This is mostly a runtime check, since type errors might be ignored at compile time.